# HG changeset patch
# User welpy-cw <h.w.forms@arcor.de>
# Date 1707947242 -3600
# Node ID 17239bec4e1cf614c571ba4647b9a8ec0c3f32de
# Parent  b5b6fa60d962b175287a35c96b86d7a7ce205898
Bug 1851975 - Get default sort for synthetic views from preferences. r=mkmelin

The default sort order and type for synthetic views such as conversations are hard coded. Use
the `mailnews.default_sort_order` and `mailnews.default_sort_type` preferences instead.

This patch also restores the missing sort indicator when a synthetic view is opened.

Differential Revision: https://phabricator.services.mozilla.com/D200067

diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -6689,16 +6689,17 @@ function restoreState({
       // So that nsMsgSearchDBView::GetHdrForFirstSelectedMessage works from
       // the beginning.
       threadTree.currentIndex = 0;
     }
 
     document.title = title;
     document.body.classList.remove("account-central");
     accountCentralBrowser.hidden = true;
+    threadPane.restoreSortIndicator();
     threadPaneHeader.onFolderSelected();
   }
 
   if (
     first &&
     messagePaneVisible &&
     Services.prefs.getBoolPref("mailnews.start_page.enabled")
   ) {
diff --git a/mailnews/db/gloda/modules/GlodaSyntheticView.jsm b/mailnews/db/gloda/modules/GlodaSyntheticView.jsm
--- a/mailnews/db/gloda/modules/GlodaSyntheticView.jsm
+++ b/mailnews/db/gloda/modules/GlodaSyntheticView.jsm
@@ -40,17 +40,26 @@ function GlodaSyntheticView(aArgs) {
   } else {
     throw new Error("You need to pass a query or collection");
   }
 
   this.customColumns = [];
 }
 GlodaSyntheticView.prototype = {
   defaultSort: [
-    [Ci.nsMsgViewSortType.byDate, Ci.nsMsgViewSortOrder.descending],
+    [
+      Services.prefs.getIntPref(
+        "mailnews.default_sort_type",
+        Ci.nsMsgViewSortType.byDate
+      ),
+      Services.prefs.getIntPref(
+        "mailnews.default_sort_order",
+        Ci.nsMsgViewSortOrder.descending
+      ),
+    ],
   ],
 
   /**
    * Request the search be performed and notification provided to
    *  aSearchListener.  If results are already available, they should
    *  be provided to aSearchListener without re-performing the search.
    */
   search(aSearchListener, aCompletionCallback) {
